named_test.go

```diff
@@ -296,3 +298,81 @@ func TestNamedQueries(t *testing.T) {
 
 	})
 }
+
+func TestNamedBulkInsert(t *testing.T) {
+	type Val struct {
+		K string `db:"k"`
+		V int    `db:"v"`
+	}
+
+	vs := []Val{{K: "x"}, {K: "y"}, {K: "z"}}
+	table := []struct {
+		values      []interface{}
+		q, expected string
+	}{
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k) VALUES (:k)",
+			expected: "INSERT INTO val (k) VALUES (?)",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k) VALUES (:k)",
+			expected: "INSERT INTO val (k) VALUES (?),(?)",
+		},
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k,v) VALUES (:k,:v)",
+			expected: "INSERT INTO val (k,v) VALUES (?,?)",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k,v) VALUES (:k,:v)",
+			expected: "INSERT INTO val (k,v) VALUES (?,?),(?,?)",
+		},
+		{
+			values:   []interface{}{vs[0]},
+			q:        "INSERT INTO val (k,v) VALUES ( :k, :v )",
+			expected: "INSERT INTO val (k,v) VALUES ( ?, ? )",
+		},
+		{
+			values:   []interface{}{vs[0], vs[1]},
+			q:        "INSERT INTO val (k,v) VALUES ( :k, :v )",
+			expected: "INSERT INTO val (k,v) VALUES ( ?, ? ),( ?, ? )",
+		},
+		{
+			values: []interface{}{vs[0], vs[1]},
+			q: func() string {
+				_, _, now := defaultSchema.Postgres()
+				return fmt.Sprintf("INSERT INTO val (k, v, added_at) VALUES (:k, :v, %v)", now)
+			}(),
+			expected: "INSERT INTO val (k, v, added_at) VALUES (?, ?, now()),(?, ?, now())",
```

Comment:
Supports only no-arguments function (e.g. `now()`).  (and include whitespace is not supported (e.g. `now(   )`))

Reply:
fixed